test/tests/functional/pbs_snapshot_unittest.py

```diff
@@ -0,0 +1,289 @@
+                for line in fd:
+                    if "job_history_duration" in line:
+                        # Remove whitespaces
+                        line = "".join(line.split())
+                        # Split it up by '='
+                        key_val = line.split("=")
+                        self.assertEquals(key_val[1], job_hist_duration)
+
+        # Cleanup
+        if os.path.isdir(snap_dir) is True:
+            self.du.rm(path=snap_dir, recursive=True, force=True)
+
+    def test_capture_all(self):
+        """
+        Test the 'capture_all' interface of PBSSnapUtils
+
+        WARNING: Assumes that the test is being run on type - 1 PBS install
+        """
+        target_dir = os.path.join(os.path.sep, "tmp")
+        num_daemon_logs = 2
+        num_acct_logs = 5
+
+        # Check that all PBS daemons are up and running
+        all_daemons_up = self.server.isUp()
+        all_daemons_up = all_daemons_up and self.mom.isUp()
+        all_daemons_up = all_daemons_up and self.comm.isUp()
+        all_daemons_up = all_daemons_up and self.scheduler.isUp()
+
+        if all_daemons_up is False:
+            # Skip the test
+            self.skipTest("Type 1 installation not present or " +
+                          "all daemons are not running")
+
+        with PBSSnapUtils(out_dir=target_dir, acct_logs=num_acct_logs,
+                          daemon_logs=num_daemon_logs, sudo=True) as snap_obj:
+            snap_dir = snap_obj.capture_all()
+            snap_obj.finalize()
+
+            # Test that all the expected information has been captured
+            # PBSSnapUtils has various dictionaries which store metadata
+            # for various objects. Create a list of these dicts
+            all_info = [snap_obj.server_info, snap_obj.job_info,
+                        snap_obj.node_info, snap_obj.comm_info,
+                        snap_obj.hook_info, snap_obj.sched_info,
+                        snap_obj.resv_info, snap_obj.datastore_info,
+                        snap_obj.pbs_info, snap_obj.core_info,
+                        snap_obj.sys_info]
+            skip_list = [ACCT_LOGS, QMGR_LPBSHOOK_OUT, "reservation", "job",
+                         QMGR_PR_OUT, PG_LOGS, "core_file_bt",
+                         "pbs_snapshot.log"]
+            platform = self.du.get_platform()
+            if platform.startswith("linux") is False:
+                skip_list.extend([ETC_HOSTS, ETC_NSSWITCH_CONF, LSOF_PBS_OUT,
+                                  VMSTAT_OUT, DF_H_OUT, DMESG_OUT])
+            for item_info in all_info:
+                for key, info in item_info.iteritems():
+                    info_path = info[0]
+                    if info_path is None:
+                        continue
+                    # Check if we should skip checking this info
+                    skip_item = False
+                    for item in skip_list:
+                        if type(item) == int:
```

Comment:
Hope this will answer your question: see 18th point at https://www.python.org/dev/peps/pep-0008/#programming-recommendations
and https://stackoverflow.com/questions/1549801/differences-between-isinstance-and-type-in-python/1549815#1549815